includes/Classifai/Providers/Watson/NLU.php

```diff
@@ -121,41 +121,45 @@ public function render_provider_fields() {
 				'input_type'    => 'password',
 				'large'         => true,
 				'class'         => 'classifai-provider-field provider-scope-' . static::ID, // Important to add this.
-				'description'   => sprintf(
-					wp_kses(
-						/* translators: %1$s is the link to register for an IBM Cloud account, %2$s is the link to setup the NLU service */
-						__( 'Don\'t have an IBM Cloud account yet? <a title="Register for an IBM Cloud account" href="%1$s">Register for one</a> and set up a <a href="%2$s">Natural Language Understanding</a> Resource to get your API key.', 'classifai' ),
-						[
-							'a' => [
-								'href'  => [],
-								'title' => [],
-							],
-						]
+				'description'   => $this->feature_instance->is_feature_enabled() ?
+					'' :
+					sprintf(
+						wp_kses(
+							/* translators: %1$s is the link to register for an IBM Cloud account, %2$s is the link to setup the NLU service */
+							__( 'Don\'t have an IBM Cloud account yet? <a title="Register for an IBM Cloud account" href="%1$s">Register for one</a> and set up a <a href="%2$s">Natural Language Understanding</a> Resource to get your API key.', 'classifai' ),
+							[
+								'a' => [
+									'href'  => [],
+									'title' => [],
+								],
+							]
+						),
+						esc_url( 'https://cloud.ibm.com/registration' ),
+						esc_url( 'https://cloud.ibm.com/catalog/services/natural-language-understanding' )
 					),
-					esc_url( 'https://cloud.ibm.com/registration' ),
-					esc_url( 'https://cloud.ibm.com/catalog/services/natural-language-understanding' )
-				),
 			]
 		);
 
-		add_settings_field(
-			static::ID . '_toggle',
-			'',
-			function ( $args = [] ) {
-				printf(
-					'<a id="classifai-waston-cred-toggle" href="#" class="%s">%s</a>',
-					$args['class'] ? esc_attr( $args['class'] ) : '', // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
-					$this->use_username_password() // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
-						? esc_html__( 'Use a username/password instead?', 'classifai' )
-						: esc_html__( 'Use an API Key instead?', 'classifai' ) // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
-				);
-			},
-			$this->feature_instance->get_option_name(),
-			$this->feature_instance->get_option_name() . '_section',
-			[
-				'class' => 'classifai-provider-field hidden provider-scope-' . static::ID, // Important to add this.
-			]
-		);
+		if ( ! $this->feature_instance->is_feature_enabled() ) {
```

Comment:
I don't think we want to hide this toggle as someone may want to switch from using an API key to using a Username and Password, which this makes that impossible unless they remove credentials first or disable the Feature first.

Reply:
Good point @dkotter, thank you. This should be fine now.